### PR TITLE
React: Fix CSF component typing

### PR DESCRIPTION
**Pull request by shilman**
Issue: #12056 

## What I did

Fixed bad component type for React

### CODE REVIEW

**Comment by shilman** [comment]
Self-merging @ndelangen @tmeasday

File: app/react/src/client/preview/types-6-0.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
-import { Component, FunctionComponent } from 'react';
+import { ComponentClass, FunctionComponent } from 'react';
 import { Args as DefaultArgs, Annotations, BaseMeta, BaseStory } from '@storybook/addons';
 import { StoryFnReactReturnType } from './types';
 
 export { Args, ArgTypes, Parameters, StoryContext } from '@storybook/addons';
 
-type ReactComponent = Component | FunctionComponent<any>;
+type ReactComponent = ComponentClass<any> | FunctionComponent<any>;
```
Comment (disintegrator) [inline app/react/src/client/preview/types-6-0.ts:7]: There's a `React.ComponentType<Props>` type that can replace this perhaps?

Comment (shilman) [reply app/react/src/client/preview/types-6-0.ts:7]: Thanks @disintegrator https://github.com/storybookjs/storybook/pull/12110

**Comment by HUYNHHOANGNAM** [comment]
Tôi đã có đủ bản ful do tôi thiết kế lập trình

Vào 19:49, Th 3, 18 thg 8, 2020 Georges Haidar <notifications@github.com>
đã viết:

**Pull request merged**